Orbitersdk/samples/ProjectApollo/src_sys/PanelSDK/Internals/Esystems.cpp

```diff
@@ -1714,4 +1714,83 @@ void Diode::Save(FILEHANDLE scn)
 
 	sprintf(cbuf, "%s", name);
 	oapiWriteScenario_string(scn, "    <DIODE> ", cbuf);
-}
\ No newline at end of file
+}
+
+//------------------------------ Light Class -----------------------------------
+electricLight::electricLight(char* lightname, e_object* i_src, const bool flashing, const double onTime, const double offTime, VESSEL *thisVessel, VECTOR3 pos, VECTOR3 dir, double range, double att0, double att1, double att2, double umbra, double penumbra, COLOUR4 diffuse, COLOUR4 specular, COLOUR4 ambient, double powerDraw, double nomVoltage)
+{
+	strcpy(name, lightname);
```

Comment:
strcpy is unsafe. If lightname is longer than 100 bytes this will cause a buffer overflow.

Reply:
We have a lot of these to change. what do you think a good alternative approach should be?